src/main/java/com/amazon/opendistroforelasticsearch/security/OpenDistroSecurityPlugin.java

```diff
@@ -1088,7 +1089,14 @@ public void onNodeStarted() {
             }
         };
     }
-    
+
+    @Override
+    public Collection<SystemIndexDescriptor> getSystemIndexDescriptors(Settings settings) {
```

Comment:
Shouldn't this be part of the OpenDistroSecuritySSLPlugin.java class ?

Could there be any issues during deployments when switching type to SecurityIndexPlugin ?

Reply:
`OpenDistroSecuritySSLPlugin.java` has default implementation of `getSystemIndexDescriptors()`, only `OpenDistroSecurityPlugin` overrides `getSystemIndexDescriptors()` to include `.opendistro_security` index.

I don't see any issue with deployments while enabling SecurityIndexPlugin, it is not a persisted settings in the metadata, so scope of the change is a single node as far as I can see.

Reply:
After some discussion, it seems that OpenDistroSecuritySSLPlugin class is used only to enable encryption-in-transit which does not use .opendistro_security index. 

Keeping the method here seems the right approach.